rllib/evaluation/rollout_worker.py

```diff
@@ -1270,22 +1279,43 @@ def add_policy(
         if policy_state:
             new_policy.set_state(policy_state)
 
-        filter_shape = tree.map_structure(
-            lambda s: (
-                None
-                if isinstance(s, (Discrete, MultiDiscrete))  # noqa
-                else np.array(s.shape)
-            ),
-            new_policy.observation_space_struct,
-        )
+        # Enabling connectors is not per policy, so we can use the general config here
+        if not connectors_enabled:
+            filter_shape = tree.map_structure(
+                lambda s: (
+                    None
+                    if isinstance(s, (Discrete, MultiDiscrete))  # noqa
+                    else np.array(s.shape)
+                ),
+                new_policy.observation_space_struct,
+            )
+
+            self.filters[policy_id] = get_filter(
+                (config or {}).get("observation_filter", "NoFilter"), filter_shape
+            )
 
-        self.filters[policy_id] = get_filter(self.observation_filter, filter_shape)
+        if connectors_enabled and policy_id in self.policy_map:
+            create_connectors_for_policy(self.policy_map[policy_id], self.policy_config)
```

Comment:
I feel like we shouldn't call create_connectors_for_policy() here, it's actually called right below, about 20 lines down.
maybe the high level flow should be:
```
if connector_enabled:
    create_connectors_for_policy(...)
...
setup_filters_if_necessary()
```
where ``setup_filters_if_necessary()`` will do:
```
def setup_filters_if_necessary(self):
    if connectors_enabled:
        self.filters[policy_id] = <try to see if we have a filter connector in the agent list>
    else:
        self.filters[policy_id] = <set up a new filter>
```
wdyt?

Reply:
Done. Like we discussed privately, we now try to create connectors for the added policy and fail with an assertion error if this policy already has connectors.